rule-engine/rule-engine-components/src/main/java/org/thingsboard/rule/engine/telemetry/TbMsgTimeseriesNode.java

```diff
@@ -120,9 +152,68 @@ public static long computeTs(TbMsg msg, boolean ignoreMetadataTs) {
         return ignoreMetadataTs ? System.currentTimeMillis() : msg.getMetaDataTs();
     }
 
+    private record PersistenceDecision(boolean saveTimeseries, boolean saveLatest, boolean sendWsUpdate) {}
+
+    private PersistenceDecision makePersistenceDecision(long ts, UUID originatorUuid) {
+        boolean saveTimeseries;
+        boolean saveLatest;
+        boolean sendWsUpdate;
+
+        if (persistenceSettings instanceof OnEveryMessage) {
+            saveTimeseries = true;
```

Comment:
Create few constants with the default values of those settings and use them. For example:
TimeseriesSaveRequestSettings.SAVE_ALL;
TimeseriesSaveRequestSettings.WS_ONLY;
TimeseriesSaveRequestSettings.SKIP;

Reply:
3 out of 4 if statements here may return a constant and no need to create a PersistenceDecision object every time.